src/likelihoods/gaussian.jl

```diff
@@ -0,0 +1,22 @@
+"""
+    GaussianLikelihood(σ²)
+
+Gaussian likelihood with `σ²` variance. This is to be used if we assume that the 
+uncertainity associated with the data follows a Gaussian distribution.
+
+```math
+    p(y|f) = Normal(y | f, σ²)
+```
+On calling, this would return a normal distribution with mean `f` and variance σ².
+"""
+struct GaussianLikelihood{T<:Real}
+    σ²::T
+end
+
+GaussianLikelihood() = GaussianLikelihood(1e-6)
+
+@functor GaussianLikelihood
+
+(l::GaussianLikelihood)(f::Real) = Normal(f, l.σ²)
+
+(l::GaussianLikelihood)(fs::AbstractVector{<:Real}) = Product([Normal(f, l.σ²) for f in fs])
```

Comment:
Maybe it might make sense to avoid the iteration and allocation of the intermediate array of distributions and just define
```suggestion
(l::GaussianLikelihood)(fs::AbstractVector{<:Real}) = MvNormal(fs, sqrt(l.σ²))
```

In general I guess one should avoid calling `Product` directly but instead use `product` (same as `Truncated` vs `truncated`) since `product` allows to define (potentially) more efficient dispatches. For instance, here `product` would create just a `MvNormal` distribution.

Reply:
I wasn't aware of the difference between `Product` and `product`. Thank you!

Reply:
Oh I just remembered that it's actually called `product_distribution` (I guess `product` would be a too generic name :smile:).